UpcomingMovies.Forms.UI/Views/MovieDetailPage.xaml.cs

```diff
@@ -0,0 +1,393 @@
+﻿using MvvmCross.Forms.Views;
+using MvvmCross.Forms.Presenters.Attributes;
+using UpcomingMovies.Core.ViewModels;
+using Xamarin.Forms;
+using FFImageLoading.Forms;
+using Xamarin.Essentials;
+using UpcomingMovies.Forms.UI.ValueConverters;
+using UpcomingMovies.Forms.UI.Effects;
+using System;
+using AiForms.Renderers;
+using UpcomingMovies.Forms.UI.Templates;
+
+namespace UpcomingMovies.Forms.UI.Views
+{
+    [MvxContentPagePresentation]
+    public partial class MovieDetailPage : MvxContentPage<MovieDetailViewModel>
+    {
+        CachedImage backdropImage;
+
+        int currentBackdropCalculatedHeight;
+        int backdropMinHeight = 200;
+        float backdropDesiredHeightPercent = 0.3f;
+
+        public MovieDetailPage()
+        {
+            InitializeComponent();
+
+            CalculateBackdropSize();
+            SetupViews();
+            SetupScroll();
+
+            SizeChanged += (sender, e) => CalculateBackdropSize();
+        }
+
+        void CalculateBackdropSize()
+        {
+            DisplayInfo info = DeviceDisplay.MainDisplayInfo;
+            double dipSize = 0;
+
+            if (info.Orientation == DisplayOrientation.Portrait)
+                dipSize = info.Height / info.Density;
+            else
+                dipSize = info.Width / info.Density;
+
+            currentBackdropCalculatedHeight = (int)(dipSize * backdropDesiredHeightPercent) <= backdropMinHeight ? backdropMinHeight : (int)(dipSize * backdropDesiredHeightPercent);
+        }
+
+        void SetupScroll()
+        {
+            scrollView.Scrolled += (sender, e) =>
+            {
+                if (e.ScrollY < 0)
+                {
+                    backdropImage.Layout(new Rectangle(e.ScrollY, e.ScrollY, scrollView.Width - e.ScrollY * 2, currentBackdropCalculatedHeight - e.ScrollY));
+                }
+                else
+                {
+                    backdropImage.Layout(new Rectangle(0, 0, scrollView.Width, backdropImage.Height));
+                }
+            };
+        }
+
+        void SetupViews()
+        {
+            SetupBackdropImage();
+
+            masterRelativeContainer.Children.Add(backdropImage,
+            xConstraint: Constraint.RelativeToParent((parent) =>
+            {
+                return parent.X;
+            }), yConstraint: Constraint.RelativeToParent((parent) =>
+            {
+                return parent.Y;
+            }), widthConstraint: Constraint.RelativeToParent((parent) =>
+            {
+                return parent.Width;
+            }), heightConstraint: Constraint.RelativeToParent((parent) =>
+            {
+                return currentBackdropCalculatedHeight;
+            }));
+
+            var infosRelativeContainer = new RelativeLayout();
```

Comment:
That's interesting approach. So do you prefer building such views this way or were there any specific reasons? I mean using `RelativeLayout` in the code vs `AbsoluteLayout` + `StackLayout` in XAML?

Reply:
At first I did try to build it with XAML, but it's hard to write RelativeLayout properly in XAML so I decided to do it in the codebehind. This page is the most important page of the app so I wanted to make sure it looked good in all screen sizes.